File: Backend/src/main/java/com/Backend/entity/Fine.java
package com.Backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "fines")
@Data
public class Fine {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String referenceNumber;
    
    private String categoryId;
    
    private String categoryName;

    @Column(columnDefinition = "numeric")
    private Double amount;

    private String violationDescription;

    private String officerName;

    private String officerBadge;

    private OffsetDateTime issuedDate;

    private String location;

    private String status; // PENDING or PAID

    private UUID driverId;

    private String driverEmail;

    private OffsetDateTime createdAt = OffsetDateTime.now();
}

File: android/traffic_fine_system/lib/data/model/fine.dart
// lib/data/model/fine.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'fine.freezed.dart';
part 'fine.g.dart';

@freezed
class Fine with _$Fine {
  @JsonSerializable(fieldRename: FieldRename.snake)
  const factory Fine({
    required String referenceNumber,
    required String categoryId,
    required String categoryName,
    required double amount,
    required String violationDescription,
    required String officerName,
    required String officerBadge,
    required String issuedDate,
    required String location,
    String? driverEmail,
    @Default('PENDING') String status,
  }) = _Fine;

  factory Fine.fromJson(Map<String, dynamic> json) => _$FineFromJson(json);
}
